src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,124 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.ColumnBuilder;
+import edu.washington.escience.myria.column.builder.ColumnFactory;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Operator which splits a string-valued column on a Java regular expression and duplicates the input row with each
+ * segment of the split result.
+ * 
+ * E.g., (1, 2, "foo:bar:baz") -> (1, 2, "foo"), (1, 2, "bar"), (1, 2, "baz")
+ */
+public class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Index of (string-valued) column to split on regex.
+   */
+  private final int splitColumnIndex;
+
+  /**
+   * Compiled regex to split input tuples.
+   */
+  private final Pattern pattern;
+
+  /**
+   * Schema of output tuples.
+   */
+  private Schema outputSchema;
+
+  /**
+   * Buffer to hold finished and in-progress TupleBatches.
+   */
+  private TupleBatchBuffer outputBuffer;
+
+  /**
+   * 
+   * @param child child operator that data is fetched from
+   * @param columnName name of string column to split using regex
+   * @param regex regular expression to split string input column on
+   */
+  public Split(final Operator child, final int splitColumnIndex, final String regex) {
+    super(child);
+    this.splitColumnIndex = splitColumnIndex;
+    pattern = Pattern.compile(regex);
+  }
+
+  private TupleBatch getChildBatch() throws DbException {
+    TupleBatch childTuples;
+    try {
+      childTuples = getChild().fetchNextReady();
+    } catch (Exception e) {
+      // only wrap checked exceptions that do not extend DbException
+      Throwables.propagateIfPossible(e, DbException.class);
+      throw new DbException(e);
+    }
+    return childTuples;
+  }
```

Comment:
(well, or `RuntimeException` but we don't want those to be wrapped in `DbException` anyway)